### PR TITLE
Added dependabot configuration

### PR DESCRIPTION
**Pull request by ribafish**
Added dependabot configuration, configured for GE and CCUD plugins, as well as for github actions.

Note: After this is merged, dependabot PRs for dependency version updates will need to be enabled in the repositories settings (if not already). You can enable it in [security settings](https://github.com/line/armeria/settings/security_analysis).

Motivation:

Dependabot will check GE and CCUD plugin versions and open PRs with updates for them. It will also check github action versions and open PRs for that.

Modifications:

- Added dependabot configuration, configured for GE and CCUD plugins, as well as for github actions.

Result:

- It will open PRs for updates to Gradle Enterprise and Common Custom User Data Gradle plugins.
- It will open PRs for updates to Github Actions used.

### CODE REVIEW

**Comment by jrhee17** [comment]
I've added this commit to my repo to check if:
- dependabot can find the dependencies well
- just to check what the pull request will look like

Let me go ahead and merge this if I don't find any issues 😄 
https://github.com/jrhee17/armeria/commits/main

**Comment by ribafish** [comment]
After this is merged, dependabot PRs for dependency version updates will need to be enabled in the repositories settings (if not already). You can enable it in [security settings](https://github.com/line/armeria/settings/security_analysis).

File: .github/dependabot.yml
Comment (ikhoon) [inline .github/dependabot.yml:15]: To avoid alarm bell at night. 😆
```suggestion
          time: "10:00"
```

File: .github/dependabot.yml
Comment (ikhoon) [inline .github/dependabot.yml:29]:
```suggestion
          time: "10:00"
```

**Review by ikhoon — APPROVED** [review]
Thanks, @ribafish! ❤️

**Review by minwoox — APPROVED** [review]
👍 👍 👍 Thanks!

**Pull request merged**
